### ISSUE TITLE
Daily Report

### ISSUE THREAD

**Issue by leaderj1001** [leaderj1001/ReinforcementLearning_Supermario#1]
하루하루 어떤 것을 했는지 기록하고 다음 할 일에 대해서 기록합니다.

**Comment by leaderj1001** [comment]
* 2018-11-08,
  - 위 문제는 fullyconnected layer output size 문제였습니다. 1024로 늘려서 사용을 했고 그와 더불어 learning rate이 너무 커서 가만히 있는 문제가 생겼습니다. 그래서 해결했고 현재 stable하게 학습되고 있습니다.

**Comment by leaderj1001** [comment]
* 2018-11-09, 
  - 아직 distributional RL을 구현 못했는데 추가할 예정입니다.

**Comment by leaderj1001** [comment]
* 2018-11-10, 
  - distributional RL(IQN)을 구현했는데 오류가 생겨서 수정해야합니다. noisy network의 output을 4096을 바꾼 뒤 학습이 매우 stable하게 되고 있습니다. 하지만 아직도 멈추는 경우가 발생합니다. 이것은 learning_rate을 좀 더 낮추거나, beta_frame_idx를 낮춰볼 예정입니다. 제가 생각하는 이유는 noisyNet을 학습하는 과정에 weight의 gradient vanishing이라고 생각합니다.

**Comment by leaderj1001** [comment]
* 2018-11-12, 
  - distributional RL(IQN) 수정 못했고, 거리와 score에 따른 reward를 어떻게 주냐에 따라 학습하는데 문제가 많이 생기고 있습니다. 학습을 더 진행해보고 IQN, A3C를 구현해볼 예정입니다. 아직까지 그냥 달리기만 하는 구간이 있습니다.

**Comment by leaderj1001** [comment]
* 2018-11-13, 
  - distributional RL(QR-DQN) 구현했고, 현재 코드 테스트 중입니다. 이로써 Rainbow DQN이 완성되었고 다음으로 학습 안정화 및 meta-learning에 대해서 공부할 예정입니다. A3C보다 먼저 meta-learning RL을 해야할 것 같습니다.

**Comment by leaderj1001** [comment]
* 2018-11-16,
  -actor-critic 구현중에 있으며, 앞서 distributional RL(QR-DQN)이 정상적으로 작동되지 않는 것 같고 actor-critic 먼저 구현한 후에 구현해볼 예정입니다.

**Comment by leaderj1001** [comment]
* 2018-11-18,
  - distributional RL(QR-DQN)을 적용시키지 않은 모델을 train시키고 있고, 만족스러운 결과는 얻지 못했습니다.
  - actor-critic 먼저 구현해두었고, meta-learning을 구현하려고 하니 actor-critic을 많이 사용하는 것같아서 actor-critic먼저 구현하였습니다.
  - A3C 알고리즘 구현할 예정입니다. 현재 문제가 되고 있는 distributional RL(QR-DQN)은 조만간 고칠 예정입니다.
  - 그리고 reward에 민감하게 반응하는 것을 보았습니다. reward에 대해서 실험을 더 진행해볼 예정입니다.

**Comment by leaderj1001** [comment]
* 2018-11-21
  - 기존에 작성했던 dqn 및 a2c, a3c를 VGG network를 구현했습니다.

**Comment by leaderj1001** [comment]
* 2018-11-22
  - soft-actor-critic 논문을 읽을려고 하니 개념적인 문제가 많아서, actor-critic 부분 부터는 강의를 듣고 시작하려고 합니다.
  - 어제 작성했던 VGG network를 실행시켜보았으나 gradient가 죽는 문제가 발생하여 학습이 되지 않아서 간소화시켰습니다.
  - '팡요랩' 강화학습 1강 강의를 들었습니다.

**Comment by leaderj1001** [comment]
* 2018-11-24
  - model free prediction, model free control을 정확히 이해하였습니다.
  - 2018-11-25에는 policy gradient를 듣고, actor critic을 구현해볼 예정입니다.

**Comment by leaderj1001** [comment]
* 2018-11-25
  - model free control, value function approximation강의를 들었습니다.

**Comment by leaderj1001** [comment]
* 2018-11-26
  - policy gradient 강의를 듣고 정리를 하였습니다.

**Comment by leaderj1001** [comment]
* 2018-12-01
  - 이미지 데이터 전처리 모듈을 작성하였습니다.
  - 4-steps 이미지 데이터를 하나의 데이터로 만드는 코드를 작성하였습니다.

**Comment by leaderj1001** [comment]
* 2018-12-02
  - a3c 알고리즘을 작성하였습니다.
  - 코드 테스트 진행해야 합니다.
